### PR TITLE
Update and clean-up dependencies, remove npm CLI dependency in scripts

### PR DESCRIPTION
**Pull request by surilindur**
Here is a set of changes to update dependencies, in addition to the following changes:

* Remove the need to have npm CLI installed to run the package scripts, by replacing `npm` with `yarn` in them.
* Remove the pre-commit hook dependency on npm CLI by using `husky` instead of `pre-commit`.
* Update the linting rules to `@rubensworks/eslint-config` version 2, and fix all the linter errors that appeared.
* Simplify the TSConfig and ESLint include/exclude listings.

Any feedback is welcome! If needed, I can also split this PR further.

### CODE REVIEW

File: tsconfig.eslint.json [hunk 1]
```diff
@@ -2,11 +2,8 @@
   "extends": "./tsconfig.json",
   "include": [
     "index.ts",
+    "bin/**/*.ts",
     "lib/**/*.ts",
-    "test/**/*.ts",
```
Comment (rubensworks) [inline tsconfig.eslint.json:6]: Does this mean tests won't get linted anymore?

**Comment by rubensworks** [comment]
Could you bump the CI to Node 16? (and perhaps also include 20)

File: tsconfig.eslint.json (hunk 1 above)
Comment (surilindur) [reply tsconfig.eslint.json:6]: The diff makes it look like the line was removed, but it is actually still there on line 7. I had to check it myself, too.

File: tsconfig.eslint.json [hunk 2]
```diff
@@ -2,11 +2,8 @@
...
-    "bin/**/*.ts"
-  ],
-  "exclude": [
-    "**/node_modules"
+    "test/**/*.ts"
```
Comment (surilindur) [inline tsconfig.eslint.json:7]: Here

Comment (rubensworks) [reply tsconfig.eslint.json:7]: Ah ok 👍

**Comment by surilindur** [comment]
This is rebased to include the changes from #19 so the tests now pass in the CI

**Comment by rubensworks** [comment]
Super, thanks!

**Pull request merged**